mbf_costmap_nav/src/mbf_costmap_nav/free_pose_search.cpp

```diff
@@ -212,64 +223,74 @@ SearchSolution FreePoseSearch::findValidOrientation(const costmap_2d::Costmap2D&
 
 SearchSolution FreePoseSearch::search() const
 {
+  // restart markers
+  if (viz_)
+  {
+    viz_->deleteMarkers();
+  }
+
   const auto costmap2d = costmap_.getCostmap();
 
   // lock costmap so content doesn't change while adding cell costs
   boost::unique_lock<costmap_2d::Costmap2D::mutex_t> lock(*(costmap2d->getMutex()));
 
   std::unordered_set<int> in_queue_or_visited;
+  std::priority_queue<Cell, std::vector<Cell>, decltype(compare_strategy_)> queue(compare_strategy_);
 
   const std::vector<geometry_msgs::Point> footprint =
       safetyPadding(costmap_, config_.use_padded_fp, config_.safety_dist);
 
-  SearchSolution sol;
-  sol.pose.theta = config_.goal.theta;
-  bool outside_or_unknown{ false };
-
-  // initializing queue with the goal cell
-  Cell start;
-  if (costmap2d->worldToMap(config_.goal.x, config_.goal.y, start.x, start.y))
-  {
-    start.cost = costmap2d->getCost(start.x, start.y);
-  }
-  else
+  // enforce bounds if goal is outside the map; and get the pose of the center of the cell
+  Cell goal_cell;
+  geometry_msgs::Pose2D goal_cell_pose;  // this is the pose of the center of the goal cell (not the goal pose)
+  int goal_cell_x, goal_cell_y;
+  costmap2d->worldToMapEnforceBounds(config_.goal.x, config_.goal.y, goal_cell_x, goal_cell_y);
+  goal_cell.x = static_cast<unsigned int>(goal_cell_x);
+  goal_cell.y = static_cast<unsigned int>(goal_cell_y);
+  goal_cell.cost = costmap2d->getCost(goal_cell.x, goal_cell.y);
+  costmap2d->mapToWorld(goal_cell.x, goal_cell.y, goal_cell_pose.x, goal_cell_pose.y);
+
+  unsigned int dummy_x, dummy_y;
+  // don't start the search from the goal pose if goal is not within bounds
+  bool test_goal_pose = costmap2d->worldToMap(config_.goal.x, config_.goal.y, dummy_x, dummy_y);
```

Comment:
```suggestion
  // don't start the search from the goal pose if goal is not within bounds
  unsigned int dummy_x, dummy_y;
  bool test_goal_pose = costmap2d->worldToMap(config_.goal.x, config_.goal.y, dummy_x, dummy_y);
```
S1: move comment up for better grouping and readability
S2: move these 2 lines below (after the if statement) in order for the logic around `goal_cell` to be grouped